src/aap_eda/api/serializers/decision_environment.py

```diff
@@ -73,9 +73,11 @@ class DecisionEnvironmentCreateSerializer(serializers.ModelSerializer):
 
     def validate(self, data):
         eda_credential_id = data.get("eda_credential_id")
+        image_url = data.get("image_url") or self.instance.image_url
         if eda_credential_id:
-            image_url = data.get("image_url") or self.instance.image_url
             validators.check_if_de_valid(image_url, eda_credential_id)
+        else:
```

Comment:
Minor: Since you are making the credential id optional in `check_if_de_valid` there is no need for the conditional here, you can simplify it just calling `validators.check_if_de_valid(image_url, eda_credential_id)` always.

Reply:
added